app/models/repository_row.rb

```diff
@@ -14,4 +16,47 @@ class RepositoryRow < ActiveRecord::Base
             presence: true,
             length: { maximum: Constants::NAME_MAX_LENGTH }
   validates :created_by, presence: true
+
+  def self.search(repository, query, page = 1, options)
+    new_query = distinct
+                .joins(:created_by)
+                .joins(
+                  "LEFT OUTER JOIN (
+                  SELECT repository_cells.repository_row_id,
+                  repository_text_values.data AS text_value,
+                  to_char(repository_date_values.data, 'DD.MM.YYYY HH24:MI')
```

Comment:
It's here just as example and likely to be changed in the future when we enable support of date cells.